src/ObjCRuntime/INativeObject.cs

```diff
@@ -20,6 +20,15 @@ static public IntPtr GetHandle (this INativeObject self)
 		{
 			return self == null ? IntPtr.Zero : self.Handle;
 		}
+
+		static public IntPtr GetNonNullHandle (this INativeObject self, string argumentName)
+		{
+			if (self == null)
+				ThrowHelper.ThrowArgumentNullException (argumentName);
```

Comment:
Separate item, but I've found the pattern of `Ex.ThrowOnNull` extremely useful in BTfS:
```
public static T ThrowOnNull<T> (T o, string name, string message = null) where T : class
{
	if (o == null)
		throw new ArgumentNullException (name ?? "<no name supplied>",
					message ?? "");
	return o;
}
```
Maybe be want to consider using it in other places too.

Reply:
This is the pattern used by the dotnet/BCL to minimize size, some details here https://referencesource.microsoft.com/#mscorlib/system/throwhelper.cs

Like @Therzok reminded me in https://github.com/xamarin/xamarin-macios/pull/8650 your pattern would not inline. Also optional arguments so add a cost to callers. Not a big deal for a tool, but that's meant to be used extensively in generated code.

I still need to measure the code size difference (IL and AOT) before moving the PR out of it's draft status. But I need to complete the generator changes first :)